tests/util/test_speed.py

```diff
@@ -0,0 +1,63 @@
+"""Test Home Assistant speed utility functions."""
+import pytest
+
+from homeassistant.const import (
+    SPEED_INCHES_PER_DAY,
+    SPEED_INCHES_PER_HOUR,
+    SPEED_KILOMETERS_PER_HOUR,
+    SPEED_METERS_PER_SECOND,
+    SPEED_MILES_PER_HOUR,
+    SPEED_MILLIMETERS_PER_DAY,
+)
+import homeassistant.util.speed as speed_util
+
+INVALID_SYMBOL = "bob"
+VALID_SYMBOL = SPEED_KILOMETERS_PER_HOUR
+
+
+def test_convert_same_unit():
+    """Test conversion from any unit to same unit."""
+    assert speed_util.convert(2, SPEED_INCHES_PER_DAY, SPEED_INCHES_PER_DAY) == 2
+    assert speed_util.convert(3, SPEED_INCHES_PER_HOUR, SPEED_INCHES_PER_HOUR) == 3
+    assert (
+        speed_util.convert(4, SPEED_KILOMETERS_PER_HOUR, SPEED_KILOMETERS_PER_HOUR) == 4
+    )
+    assert speed_util.convert(5, SPEED_METERS_PER_SECOND, SPEED_METERS_PER_SECOND) == 5
+    assert speed_util.convert(6, SPEED_MILES_PER_HOUR, SPEED_MILES_PER_HOUR) == 6
+    assert (
+        speed_util.convert(7, SPEED_MILLIMETERS_PER_DAY, SPEED_MILLIMETERS_PER_DAY) == 7
+    )
+
+
+def test_convert_invalid_unit():
+    """Test exception is thrown for invalid units."""
+    with pytest.raises(ValueError):
+        speed_util.convert(5, INVALID_SYMBOL, VALID_SYMBOL)
+
+    with pytest.raises(ValueError):
+        speed_util.convert(5, VALID_SYMBOL, INVALID_SYMBOL)
+
+
+def test_convert_nonnumeric_value():
+    """Test exception is thrown for nonnumeric type."""
+    with pytest.raises(TypeError):
+        speed_util.convert("a", SPEED_KILOMETERS_PER_HOUR, SPEED_MILES_PER_HOUR)
+
+
+@pytest.mark.parametrize(
+    "from_value, from_unit, expected, to_unit",
+    [
+        (5, SPEED_KILOMETERS_PER_HOUR, 3.10686, SPEED_MILES_PER_HOUR),
+        (5, SPEED_MILES_PER_HOUR, 8.04672, SPEED_KILOMETERS_PER_HOUR),
+        (5, SPEED_INCHES_PER_DAY, 127, SPEED_MILLIMETERS_PER_DAY),
+        (5, SPEED_MILLIMETERS_PER_DAY, 0.19685, SPEED_INCHES_PER_DAY),
+        (5, SPEED_INCHES_PER_DAY, 127, SPEED_MILLIMETERS_PER_DAY),
+        (5, SPEED_METERS_PER_SECOND, 708661, SPEED_INCHES_PER_HOUR),
+        (5000, SPEED_INCHES_PER_HOUR, 0.03528, SPEED_METERS_PER_SECOND),
```

Comment:
Ah I see how we are looking at this differently. I got all of these numbers straight from Google (e.g. `5 in per day to mm per day`, so figured using the magic numbers rather than calculations would be easier to hand check since I can plug everything into the search.

I can add the calculations as well though like you described!